in_toto/models/metadata.py

```diff
@@ -198,6 +256,48 @@ def sign_gpg(self, gpg_keyid=None, gpg_home=None):
     return signature
 
 
+  def verify(self, key: Key) -> bool:
+    """Verifies a signature over signable in signature with verification key.
+
+    Uses the UTF-8 encoded canonical JSON byte representation of the signable
+    attribute to verify the signature deterministically.
+
+    Raises:
+      securesystemslib.exceptions.FormatError: The passed key is malformed.
+
+      SignatureVerificationError: No signature keyid matches the verification
+          key keyid, or the matching signature is malformed, or the matching
+          signature is invalid.
+
+      securesystemslib.gpg.exceptions.KeyExpirationError: Passed verification
+          key is an expired gpg key.
+
+    """
+
+    for signature in self.signatures:
+      if key.match_keyid(signature['keyid']):
+        break
+
+    else:
+      raise SignatureVerificationError("No signature found for key '{}'"
+          .format(key.keyid))
+
+    if securesystemslib.formats.GPG_SIGNATURE_SCHEMA.matches(signature):
+      valid = key.verify(
+        GPGSignature.from_dict(signature), self.signed.signable_bytes)
+
+    elif securesystemslib.formats.SIGNATURE_SCHEMA.matches(signature):
+      valid = key.verify(
+        Signature.from_dict(signature), self.signed.signable_bytes)
```

Comment:
Ideally, `Metablock.signatures` is already a list of `Signature` objects, so you don't have to case handle here. Can you add it to the issue with `Metablock` enhancements that may or may not be worth providing before we deprecate it?

Reply:
`Metablock.signatures` is not a list of `Signature` objects right now. Also, I found a flaw in the verify method added previously, `Signature.from_dict` destroys the dict of signature provided.

The solution to add a method similar to the DSSE having same arguments would be tricky. The one way I can think of is to convert the whole Metablock wrapper to use `Signer`, `Key` and `Signature` as objects.

Right now I removed the method and using the `Metablock` verify signature method for verification.